### PR TITLE
Make `bottommost_temperature` dynamically changeable

### PR DESCRIPTION
**Pull request by jay-zhuang**
Summary: Make `AdvancedColumnFamilyOptions.bottommost_temperature`
dynamically changeable with `SetOptions` API.

Test Plan: added unittest

### CODE REVIEW

**Review by zhichao-cao — COMMENTED** [review]
One general question, when we assign the temperature to the bottom-most level, is it still under the DB mutex protection?

**Comment by jay-zhuang** [comment]
> One general question, when we assign the temperature to the bottom-most level, is it still under the DB mutex protection?

This fix is only for updating the option, assigning the temperature only happens in the next compaction:
https://github.com/jay-zhuang/rocksdb/commit/7c446412e4372218fdf87144aa206abe3bf33fe8#diff-d737a8ca0e8d739e33c030d56fdfdb5da3c54a2772750eef1c3a7f022e2a54d3R6732-R6747

**Comment by zhichao-cao** [comment]
> > One general question, when we assign the temperature to the bottom-most level, is it still under the DB mutex protection?
> 
> This fix is only for updating the option, assigning the temperature only happens in the next compaction: [jay-zhuang@7c44641#diff-d737a8ca0e8d739e33c030d56fdfdb5da3c54a2772750eef1c3a7f022e2a54d3R6732-R6747](https://github.com/jay-zhuang/rocksdb/commit/7c446412e4372218fdf87144aa206abe3bf33fe8#diff-d737a8ca0e8d739e33c030d56fdfdb5da3c54a2772750eef1c3a7f022e2a54d3R6732-R6747)

I see, so even when we call "SetOption" during compaction, current compaction will not be influenced. Thanks!

File: .circleci/config.yml [hunk 1]
```diff
@@ -162,7 +162,7 @@ executors:
 jobs:
   build-macos:
     macos:
-      xcode: 11.3.0
+      xcode: 12.5.1
```
Comment (siying) [inline .circleci/config.yml:165]: Is it related?

**Review by siying — COMMENTED** [review]
Thank you for working on it. Can you add test scenarios in option_test and options_settable_test to cover this new option?

File: .circleci/config.yml (hunk 1 above)
Comment (jay-zhuang) [reply .circleci/config.yml]: removed

**Comment by jay-zhuang** [comment]
> Thank you for working on it. Can you add test scenarios in option_test and options_settable_test to cover this new option?

Sure, updated the tests to include that.

**Review by siying — APPROVED** [review]
Thanks for the improvement!

**Pull request closed without merging**